packages/web-components/@sfx/search-box/test/unit/search-box.test.ts

```diff
@@ -0,0 +1,193 @@
+        detail: 'a',
+        bubbles: true
+      });
+      searchbox.value = 'a';
+
+      searchbox.emitSearchEvent();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchRequestEvent);
+    });
+  });
+
+  describe('emitSearchBoxClearClick', () => {
+    it('should dispatch an emitSearchBoxClearClick', () => {
+      const searchboxClearedEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK);
+
+      searchbox.emitSearchBoxClearClick();
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxClearedEvent);
+    });
+  });
+
+  describe('updateText', () => {
+    it('should update the value property of the input element in response to data received', () => {
+      const detail = 'inputText';
+      const inputObject = { value: 'Placeholder text' };
+      const inputEvent = new CustomEvent(SEARCHBOX_EVENT.UPDATE_SEARCH_TERM, { detail: detail });
+      stub(searchbox, 'getInputElement').returns(inputObject);
+
+      searchbox.updateText(inputEvent);
+
+      expect(searchbox.value).to.equal(detail);
+      expect(searchbox.value).to.equal(inputObject.value);
+    });
+  });
+
+  describe('handleKeydown', () => {
+    it('should invoke the emitSearchEvent function if enter is pressed and value property length is greater than 0', () => {
+      const emitSearchStub = stub(searchbox, 'emitSearchEvent');
+      searchbox.value = 'hello';
+
+      searchbox.handleKeydown({ key: "Enter" });
+
+      expect(emitSearchStub).to.have.been.called;
+    });
+  });
+
+  describe('handleInput', () => {
+    it('should invoke the updateSearchTermValue function', () => {
+      const updateSearchTermValueStub = stub(searchbox, 'updateSearchTermValue');
+      const target = { value: 'dee' };
+
+      searchbox.handleInput({ target });
+
+      expect(updateSearchTermValueStub).to.have.been.calledWith('dee');
+    });
+
+    it('should dispatch a search box change event', () => {
+      const target = { value: 'dee' };
+      const searchboxChangeEvent = new CustomEvent(SEARCHBOX_EVENT.SEARCHBOX_CHANGE, {
+        detail: target,
+        bubbles: true
+      });
+      stub(searchbox, 'updateSearchTermValue');
+
+      searchbox.handleInput({ target });
+
+      expect(searchboxDispatchEvent).to.have.been.calledWith(searchboxChangeEvent);
+    });
+  });
+
+  describe('updateSearchTermValue', () => {
+    it('should set the search term property to the input value', () => {
+      searchbox.updateSearchTermValue('catfood');
+
+      expect(searchbox.value).to.equal('catfood');
+    });
+  });
+
+  describe('clearSearch', () => {
+    it('should set the search term property to an empty string', () => {
+      const inputObject = { value: 'Placeholder text' };
+      stub(searchbox, 'getInputElement').returns(inputObject);
+      stub(searchbox, 'emitSearchBoxClearClick');
+
+      searchbox.clearSearch();
+
+      expect(searchbox.value).to.equal('');
+      expect(searchbox.value).to.equal(inputObject.value);
```

Comment:
If you want to assert that `inputObject.value` should be an empty string, then simply assert that.